crypto/fipsmodule/evp/evp_ctx.c

```diff
@@ -438,6 +439,50 @@ int EVP_PKEY_keygen_init(EVP_PKEY_CTX *ctx) {
   return 1;
 }
 
+int EVP_PKEY_keygen_deterministic(EVP_PKEY_CTX *ctx,
+                                  EVP_PKEY **out_pkey,
+                                  const uint8_t *seed,
+                                  size_t *seed_len) {
+  // We have to avoid potential underlying services updating the indicator state,
+  // so we lock the state here.
+  FIPS_service_indicator_lock_state();
+  int ret = 0;
+  if (!ctx || !ctx->pmeth || (!ctx->pmeth->keygen && !ctx->pmeth->keygen_deterministic)) {
```

Comment:
why do we check for `!ctx->pmeth->keygen`?

Reply:
good spot! An artifact from when I was trying to implement the rand/de-rand functionality at the EVP layer. removed in [52aa448](https://github.com/aws/aws-lc/pull/1578/commits/52aa448e04a98a3b88a426b910496451c8714712)